File: src/main/java/za/co/ioagentsmith/game/kalaha/controller/PlayerSetupController.java
package za.co.ioagentsmith.game.kalaha.controller;

import za.co.ioagentsmith.game.kalaha.model.PlayerNames;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.servlet.ModelAndView;

@Controller
public class PlayerSetupController {

    @RequestMapping(value = "/playerSetup.do", method = RequestMethod.GET)
    public ModelAndView showForm() {
        return new ModelAndView("playerSetup", "playerNames", new PlayerNames());
    }

}

File: kalaha-kotlin-backend/src/main/java/za/co/ioagentsmith/game/kalaha/controller/PlayerSetupController.kt
package za.co.ioagentsmith.game.kalaha.controller

import za.co.ioagentsmith.game.kalaha.model.PlayerNames
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod
import org.springframework.web.servlet.ModelAndView

@Controller
class PlayerSetupController {

    @RequestMapping(value = ["/playerSetup.do"], method = [RequestMethod.GET])
    fun showForm(): ModelAndView {
        return ModelAndView("playerSetup", "playerNames", PlayerNames())
    }

}